rfcs/react-components/non-core-components.md

```diff
@@ -0,0 +1,388 @@
+library but other packages. We will have to finalize the following requirements:
+
+- Should non-core components be able to deviate from V9 core concepts?
+- Should non-core components have a faster breaking change cadence?
+- Should non-core components live in a separate repo?
+- Should non-core components still be under the `@fluentui` scope?
+- How should users consume non-core packages?
+- How should non-core components be documented?
+- How should code for partner owned components be hosted?
+
+The Proposal treats each of the above questions separately and provides pros/cons for different approaches we could take.
+Keep in mind that some decisions will influence other decisions.
+
+None of the approaches are particularly complex to understand, but the resulting work could be complex. It is important
+that we figure out what answers for these questions so that we can start to move forward
+with technical implementation by following concrete requirements.
+
+## Detailed Design or Proposal
+
+### Deviating from v9 core concepts
+
+It's difficult to formalize what 'deviating' means here, and any attempt to do so will probably not be
+very future proof as the library continues to evolve. The main objective of this section is to decide whether or not
+we should allow deviations from v9 core concepts (whatever they are or might be in the future). Here are some
+examples of current deviations that could happen based on the current situation:
+
+- Allowing shorthand collections
+- 'Fatter' JSX elements that have more functionality
+- Less customizable styles
+- More opinionated layout
+
+The short list above indicates some possibilities to have components that are less flexible in cases where
+components are more specialized versions of core primitives that we don't want to maintain a large customization
+feature set for, or product/scenario specific components where we want more alignment across Microsoft.
+
+The extent of 'deviating' will have to depend on the requirements of each component, but we should be ready to accept
+that we might end up delivering a set of components that don't follow the principles that drive the core library
+(@fluentui/react-components).
+
+In cases where components need to deviate, we should notify the team through specs and architecture meetings
+(i.e. tech sync), what deviations are done and the context behind them. All non-core components should still
+follow the V9 spec driven approach.
+
+#### Pros
+
+- 👍 Can enforce design across Microsoft more strictly
+- 👍 Can be more flexible than core components
+- 👍 Can be done on a case-by-case basis depending on requirements
+
+#### Cons
+
+- 👎 Can lead to a very inconsistent set of components that behave differently to each other
+- 👎 Our partners might still want to customize to the max
+- 👎 Approaches could be heavily biased on a subset of partners
+
+### Breaking change cadence
```

Comment:
- breaking changes in none  FUI family extensions can happen as their maintainer need
- breaking changes in FUI family extensions if possible, should adopt patterns we "plan" to adopt within core -> major version bump is not really a breaking as we maintain old apis and remove those after particular major bumps or defined time period